core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -73,6 +75,16 @@ private Mappers() {
                 classLoader = Mappers.class.getClassLoader();
             }
 
+            ServiceLoader<T> loader = ServiceLoader.load( clazz, classLoader );
```

Comment:
Sure I also though about implementing a cache to let Mappers keep track of what implementation needs to be used but I think we should do some benchmarks first